mitosheet/src/mito/components/taskpanes/Graph/GraphSidebar.tsx

```diff
@@ -99,6 +160,76 @@ const GraphSidebar = (props: {
             }
             const executeScript = new Function(graphOutput.graphScript);
             executeScript()
+            const div: any = document.getElementById(graphOutput.graphHTML.split('id="')[1].split('"')[0])
+            if (div === null) {
+                return;
+            }
+            div.on('plotly_click', () => {
+                setSelectedGraphElement(null)
+            });
+            
+            // Main Title
+            const gtitle = div.getElementsByClassName('g-gtitle')[0]
+            const xtitle = div.getElementsByClassName('g-xtitle')[0]
+            const ytitle = div.getElementsByClassName('g-ytitle')[0]
+                            
+            // First, add the style to make it clickable with pointer-events: all
+            gtitle.style.pointerEvents = 'all'
+            xtitle.style.pointerEvents = 'all'
+            ytitle.style.pointerEvents = 'all'
+
+            /**
+             * Set selected graph element when clicked
+             */
+            gtitle.addEventListener('click', () => {
+                setSelectedGraphElement({
+                    element: 'gtitle',
+                    displayPopup: false
+                })
+            })
+            xtitle.addEventListener('click', () => {
+                setSelectedGraphElement({
+                    element: 'xtitle',
+                    displayPopup: false
+                });
+            })
+            ytitle.addEventListener('click', () => {
+                setSelectedGraphElement({
+                    element: 'ytitle',
+                    displayPopup: false
+                })
+            })
+
+            /**
+             * Open popup when double clicked
+             */
+            gtitle.addEventListener('dblclick', () => {
+                setSelectedGraphElement({
+                    element: 'gtitle',
+                    xPosition: gtitle.getBoundingClientRect().left,
+                    yPosition: gtitle.getBoundingClientRect().top + 30,
+                    displayPopup: true
+                })
+            });
+
+            xtitle.addEventListener('dblclick', () => {
+                const clientRect = xtitle.getBoundingClientRect()
+                setSelectedGraphElement({
+                    element: 'xtitle',
+                    xPosition: (clientRect.left + clientRect.right) / 2 - 70,
+                    yPosition: xtitle.getBoundingClientRect().top - 48,
+                    displayPopup: true
+                })
+            });
+
+            ytitle.addEventListener('dblclick', () => {
+                setSelectedGraphElement({
+                    element: 'ytitle',
+                    xPosition: ytitle.getBoundingClientRect().left - 10,
+                    yPosition: ytitle.getBoundingClientRect().top - 40,
+                    displayPopup: true
+                })
+            });
```

Comment:
Do we need to do any cleanup on these event listeners? Or are they cleaned up when the svg disappears? I have a feeling we need to do some cleanup.

Reply:
Ok so I'm not totally sure on the answer to this but it does help that we're using a unique ID for the graph. I did a quick test just now and it doesn't look like it calls the event handlers any extra times after it re-registers a new event handler.